xbmc/pvr/windows/GUIEPGGridContainer.cpp

```diff
@@ -695,7 +697,12 @@ void CGUIEPGGridContainer::UpdateItems()
         newBlockIndex = eventOffset;
       }
       else
-        newBlockIndex = (eventStart - gridStart).GetSecondsTotal() / 60 / CGUIEPGGridContainerModel::MINSPERBLOCK + eventOffset;
+      {
+        // first block of previously selected tag is always the calculated block rounded up.
+        // refer to CGUIEPGGridContainerModel::Refresh, where the model is created, for details!
+        float fBlockIndex = (eventStart - gridStart).GetSecondsTotal() / 60.0f / CGUIEPGGridContainerModel::MINSPERBLOCK;
```

Comment:
Not in this context, as here it's guaranteed that `gridStart < eventStart` => https://github.com/xbmc/xbmc/pull/12707/files#diff-7e7416a942af24ccc1256612c5ace9f4R694 ff